lib/secrets.js

```diff
@@ -30,7 +30,18 @@ function find(table, name, options, done) {
   }).query(params, done);
 }
 
-function map(name, data, done) {
+function list(table, options, done) {
+  var params = {
+    TableName: table || 'credential-store',
+    ConsistentRead: true,
+  };
+
+  return new AWS.DynamoDB({
+    httpOptions: { agent: agent }
```

Comment:
Why the addition of the `httpOptions` here? It isn't used elsewhere in calls to AWS.
I'm just curious what the advantage is.

Also, `agent` isn't defined by default or configured elsewhere prior to this reference AFAICT.